### DISCUSSION TITLE
In-Line Twist for Corkscrew Coaster (sprites already exist)

### DISCUSSION THREAD

**Discussion by CSEverett** [OpenRCT2/OpenRCT2#16132] [Feature requests] [answered]
Normally, adding new track pieces to an already existing coaster type is impractical, due to the extreme difficulty of getting the new graphics to be an exact match. The exception if it the track sprites already exist, either unused or an another coaster with identical track (such as adding elements from the Twister Coaster to the Vertical Drop Coaster).  That's actually the case with adding an In-Line Twist to the Corkscrew Coaster. The Lay-Down Coaster actually has perfectly identical track, and it includes an In-Line Twist. So adding one to the Corkscrew Coaster should be a perfectly straightforward matter, since no new sprites are needed. It would be no different than adding the extra track pieces to the Vertical Drop Coaster, except less work due to the reduced number of pieces.

**Comment by ppamorim** [comment]
This is possible but does it make sense? I think angled corkscrew could be better.

**Reply by Legotown55** [reply]
What about large loops on the corkscrew coaster and the heelix on the water slide

**Reply by ocalhoun6** [reply]
@Legotown55 

I don't think we have sprites for either of those, do we?

There aren't any coasters that I'm aware of that have corkscrew type track and have large half loops. All the ride with large half loops use the twister coaster track style. And the water slide has completely unique track, which probably doesn't include a hitherto unused helix sprite.

@ppamorim 

In-line twist probably shouldn't be enabled for the corkscrew coaster by default, but it should be made available through the 'enable all drawable track pieces' cheat.

**Reply by CSEverett** [reply]
"Enable all track pieces" sounds about right.

**Reply by Legotown55** [reply]
What about the large loop

**Reply by X123M3-256** [reply]
> What about the large loop

There are no large loop sprites for the corkscrew track in the game.

**Comment by Gymnasiast** [comment]
> The Lay-Down Coaster actually has perfectly identical track, and it includes an In-Line Twist.

The inversions on reversible roller coasters (Flying, Lay-down, Multidim) are not quite the same as those on regular ones. As such, it’s quite possible it will not align with the other inversions. Haven’t tried it, though. If it doesn’t fit, the SLC corkscrew might.

**Reply by karst** [reply]
The lay-down rollercoaster seems to line up perfectly.
![image](https://user-images.githubusercontent.com/1516893/145733815-0dde9723-8e37-497a-961a-b587ae66955d.png)
![image](https://user-images.githubusercontent.com/1516893/145733962-924f719e-21eb-4d11-97a8-4a4987549c7a.png)

The SLC does not.
![image](https://user-images.githubusercontent.com/1516893/145733826-b749d526-71d5-4446-a118-da192614a7aa.png)

**Reply by karst** [reply]
3 hours of trial and error and learning to code later. I got this in the game by adding a whole bunch of lines. I'll fix the supports tomorrow =3= 
![image](https://user-images.githubusercontent.com/1516893/145743886-4ff34449-0860-4766-b598-85843af6efb5.png)

**Comment by karst** [comment, marked as answer]
This can be marked as Answered as this has been merged into develop with #16224
